### PR TITLE
Add buf beta price command

### PR DESCRIPTION
**Pull request by bufdev**

### CODE REVIEW

File: private/buf/cmd/buf/command/beta/price/price.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+		Stats:    stats,
+		NumTypes: stats.NumMessages + stats.NumEnums + stats.NumMethods,
+	}
+	buckets := tmplData.NumTypes / 100
+	if tmplData.NumTypes%100 != 0 {
+		tmplData.ChargeableIsRounded = true
+		buckets++
+	}
+	tmplData.ChargeableTypes = buckets * 100
+	tmplData.TeamsDollarsPerMonth = fmt.Sprintf("%.2f", float64(tmplData.ChargeableTypes)*teamsPerTypeDollarsPerMonth)
```
Comment (joshcarp) [inline private/buf/cmd/buf/command/beta/price/price.go:182]: Not to be over engineering but if we do any more advanced stuff with money calculations we should probably use a decimal/money package as to not have rounding errors down the line

**Review by joshcarp — APPROVED** [review]

**Review by timostamm — APPROVED** [review]

File: private/buf/cmd/buf/command/beta/price/price.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	"github.com/spf13/pflag"
+)
+
+const (
+	disableSymlinksFlagName     = "disable-symlinks"
+	teamsPerTypeDollarsPerMonth = float64(0.50)
+	proPerTypeDollarsPerMonth   = float64(1)
+	tmplCopy                    = `Current BSR pricing:
+
+  - Teams: $50/month per 100 types
```
Comment (saquibmian) [inline private/buf/cmd/buf/command/beta/price/price.go:39]: The [pricing page](https://buf.build/pricing/) refers to this as `Team` (no plural).

File: private/buf/cmd/buf/command/beta/price/price.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+Your sources have:
+
+  - {{.NumMessages}} messages
+  - {{.NumEnums}} enums
+  - {{.NumMethods}} methods
+
+This adds up to {{.NumTypes}} types.{{if .ChargeableIsRounded}}
+
+We bill in increments of 100 types. Types are summed up across
```
Comment (saquibmian) [inline private/buf/cmd/buf/command/beta/price/price.go:53]: nit: 
```suggestion
We bill in increments of 100 types. Types are totaled across
```

File: private/buf/cmd/buf/command/beta/price/price.go (hunk 2 above)
Comment (saquibmian) [inline private/buf/cmd/buf/command/beta/price/price.go:39]: Can you indicate that this is USD?

Comment (bufdev) [reply private/buf/cmd/buf/command/beta/price/price.go:39]: That's assumed by everyone reading from a tool like this, seems superfluous

Comment (jhump) [inline private/buf/cmd/buf/command/beta/price/price.go:39]: This information seems like it should be downloaded from the server. If it's built-in to the CLI, then someone on an old version of the CLI cannot get accurate pricing projections in the face of any tweaks or changes to pricing in the future.

Maybe a better way to do this is to send all of the stats to a simple BSR endpoint to compute the estimated price?

Comment (bufdev) [reply private/buf/cmd/buf/command/beta/price/price.go]: Yea that's likely a better way, but that's more time than we want to allocate to this now, this gets this across the line and out there in under an hour's worth of work. We mention that the pricing information may not be the latest in the copy.

**Pull request merged**